File: integration-tests/opentelemetry-redis-instrumentation/src/main/java/io/quarkus/io/opentelemetry/jackson/RegisterCustomModuleCustomizer.java
package io.quarkus.io.opentelemetry.jackson;

import jakarta.inject.Singleton;

import io.quarkus.jackson.JsonMapperBuilderCustomizer;
import io.vertx.redis.client.impl.types.ErrorType;
import tools.jackson.databind.json.JsonMapper;
import tools.jackson.databind.module.SimpleModule;

@Singleton
public class RegisterCustomModuleCustomizer implements JsonMapperBuilderCustomizer {

    public void customize(JsonMapper.Builder builder) {
        SimpleModule module = new SimpleModule();
        module.addSerializer(ErrorType.class, new ErrorTypeSerializer());
        builder.addModule(module);
    }
}

File: integration-tests/resteasy-reactive-kotlin/standard/src/main/kotlin/io/quarkus/it/resteasy/reactive/kotlin/RegisterCustomModuleCustomizer.kt
package io.quarkus.it.resteasy.reactive.kotlin

import io.quarkus.jackson.JsonMapperBuilderCustomizer
import io.quarkus.runtime.annotations.StaticInitSafe
import jakarta.inject.Singleton
import org.eclipse.microprofile.config.inject.ConfigProperty
import tools.jackson.databind.json.JsonMapper

@Singleton
class RegisterCustomModuleCustomizer : JsonMapperBuilderCustomizer {
    @StaticInitSafe @ConfigProperty(name = "test.prop") lateinit var testProp: String

    override fun customize(builder: JsonMapper.Builder) {
        GreetingResource.MY_PROPERTY.set(testProp)
    }
}
